Fix merging consecutive sequences causing data loss

excelsior-dev opened the pull request:
Hello,
I tried using listFix and after a few days of finding it great I discovered data loss bug. I decided to do my first ever pull request so bear with me please. 

It led me to lizzy being the culprit. Having more than one sequence of repeats (duplicates) in a playlist (I used m3u8) was overwriting entries which are certain number of places after that sequence. It was caused by wrong indexing in "mergeConsecutiveIdenticalMedia" method. Made a small test file also, hope it's all good.

Tell me If I'm doing something wrong or it could be done better. Waiting for next listFix update!

excelsior-dev committed:
Fix merging consecutive sequences causing data loss. Additionally test was added.

Borewit reviewed (changes requested):
Code style issues: `./gradlew check :checkstyleTest `is not passing.

Awesome work @excelsior-dev . Very nice catch!
And av very brave step! You are now an open source contributor.

Next time you create a PR, **create new branch**!

If you add changes to the default (`main` on this repo) branch , you cannot sync back changes I have done the the original main branch.

That is critical cycle to solve merge conflicts. It also makes it harder to checkout your branch as now I have 2 main branches in my environment.

excelsior-dev committed:
Requested changes fix

-Fixed typo
-Using diamond when instantiating generic lists
-Ensuring to remove tmp file
-Cleaning up

excelsior-dev commented:
Thank you for the fast response!

Will create a new branch next time, I guess it will pass this time because there are no commits in original main?

Think I fixed everything you requested. Wasn't sure what's the best way to do it since there are buttons to commit suggestions you made. I chose to make changes locally and then push.

Let me know if there is anything else.

Borewit reviewed (approved):
LGTM

Borewit commented:
Part of [release v4.1.2](https://github.com/Borewit/lizzy/releases/tag/v4.1.2)
